tests/ha/test_control_pod_restart.py

```diff
@@ -383,3 +401,286 @@ def test_failover_control_pod_kubectl(self):
+                                                   self.control_node])
+        LOGGER.debug("Fail over node is: %s", failover_node)
+
+        event.set()
+        LOGGER.info("Step 6: Failover control pod %s to node %s and check cluster status",
+                    self.control_pod_name, failover_node)
+        pod_yaml = {self.control_pod_name: self.modified_yaml}
+        resp = self.ha_obj.failover_pod(pod_obj=self.node_master_list[0], pod_yaml=pod_yaml,
+                                        failover_node=failover_node)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 6: Successfully failed over control pod to %s. Cluster is in good state",
+                    failover_node)
+
+        event.clear()
+        self.restore_pod = True
+        LOGGER.info("Step 7: Verify if IAM users %s are persistent across control pod failover",
+                    uids)
+        for user in uids:
+            resp = self.rest_iam_user.get_iam_user(user)
+            assert_utils.assert_equal(resp.status_code, const.Rest.SUCCESS_STATUS,
+                                      f"Couldn't find user {user} after control pod failover")
+            LOGGER.info("User %s is persistent: %s", user, resp)
+        LOGGER.info("Step 7: Verified all IAM users %s are persistent across control pod "
+                    "failover", uids)
+
+        LOGGER.info("Waiting for background IOs thread to join")
+        thread_wr.join()
+        thread_rd.join()
+        thread_del.join()
+        LOGGER.info("Step 8.1: Verify status for In-flight DELETEs during control pod "
+                    "failover to %s", failover_node)
+        del_resp = tuple()
+        while len(del_resp) != 2:
+            del_resp = del_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not del_resp:
+            assert_utils.assert_true(False, "Background process failed to do deletes")
+        fail_del_bkt = del_resp[1]
+        rem_bkts_aftr_del = s3_test_obj.bucket_list()[1]
+        assert_utils.assert_false(len(fail_del_bkt),
+                                  f"Bucket deletion failed when cluster was online {fail_del_bkt}")
+        assert_utils.assert_equal(len(rem_bkts_aftr_del), remaining_bkt,
+                                  f"{del_bucket} buckets should get deleted, only "
+                                  f"{wr_bucket - len(rem_bkts_aftr_del)} were deleted")
+        LOGGER.info("Step 8.1: Verified status for In-flight DELETEs during control pod "
+                    "failover to %s", failover_node)
+
+        LOGGER.info("Step 8.2: Verify status for In-flight WRITEs during control pod "
+                    "failover to %s", failover_node)
+        wr_resp = dict()
+        while len(wr_resp) != 2:
+            wr_resp = wr_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        pass_logs = list(x[1] for x in wr_resp["pass_res"])
```

Comment:
Please add logger for these two list. It will help in debugging failures.

Reply:
In case of failure we are already printing list of logs which have issues.

Reply:
YEs. But I have observed, that when we are getting IOs stuck, There is not at all any log is coming. Which was not expectable as that will also give len() 0. Which is not our expectation. This was added in that scenario only. Better we should add it.